src/main/java/seedu/programmer/model/student/UniqueStudentList.java

```diff
@@ -41,15 +41,81 @@
     public boolean contains(Student toCheck) {
         requireNonNull(toCheck);
         return internalList.stream().anyMatch(toCheck::isSameStudent);
+
+    }
+
+    /**
+     * Returns true if the list contains an equivalent student as the given argument.
+     */
+    public boolean containsSameEmail(Student toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameStudentEmail);
     }
 
+
+    /**
+     * Returns true if the list contains an equivalent student as the given argument.
+     */
+    public boolean containsSameStudentId(Student toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameStudentId);
+    }
+
+
+    /**
+     * Returns true if the list contains an equivalent student as the given argument，
+     * excluding himself/herself.
+     */
+    public boolean containsOther(Student studentToEdit, Student editedStudent) {
+        requireAllNonNull(studentToEdit, editedStudent);
+        List<Student> studentListCopy = new ArrayList<>();
+        for (Student student : internalList) {
+            studentListCopy.add(student.copy());
+        }
+        studentListCopy.remove(studentToEdit);
+        return studentListCopy.stream().anyMatch(editedStudent::isSameStudent);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent student as the given argument，
+     * excluding himself/herself.
+     */
+    public boolean containsOtherSameEmail(Student studentToEdit, Student editedStudent) {
+        requireAllNonNull(studentToEdit, editedStudent);
+        List<Student> studentListCopy = new ArrayList<>();
+        for (Student student : internalList) {
+            studentListCopy.add(student.copy());
+        }
+        studentListCopy.remove(studentToEdit);
+        return studentListCopy.stream().anyMatch(editedStudent::isSameStudentEmail);
+    }
+
+
+    /**
+     * Returns true if the list contains an equivalent student as the given argument，
+     * excluding himself/herself.
+     */
+    public boolean containsOtherSameStudentId(Student studentToEdit, Student editedStudent) {
+        requireAllNonNull(studentToEdit, editedStudent);
+        List<Student> studentListCopy = new ArrayList<>();
+        for (Student student : internalList) {
+            studentListCopy.add(student.copy());
+        }
+        studentListCopy.remove(studentToEdit);
+        return studentListCopy.stream().anyMatch(editedStudent::isSameStudentId);
+    }
+
+
     /**
      * Adds a student to the list and the list is sorted each time.
      * The student must not already exist in the list.
      */
     public void add(Student toAdd) throws DuplicateStudentException {
         requireNonNull(toAdd);
-        toAdd.setLabResultRecord(labsTracker);
+        if (toAdd.getLabList().isEmpty()) {
```

Comment:
I think we can consider creating a `hasNoLabs` method in the student class which returns whether or not their list of labs is empty to avoid violating Demeter's Law.

Reply:
good suggestion!